### ISSUE TITLE
LootTableEvent seems not working properly

### ISSUE THREAD

**Issue by Stein-N** [neoforged/NeoForge#428]
**Minecraft Version:** 1.20.2

**NeoForge Version:** 20.2.86

**Logs:** no logs produced

**Steps to Reproduce:**
 1. Try to add Items to LootTables or LootPools via the LootTableEvent

## LootModifier Class
````java
@Mod.EventBusSubscriber(modid = Constants.MOD_ID, bus = Mod.EventBusSubscriber.Bus.FORGE)
public class LootTableModifier {

    private static LootTable table;
    private static ResourceLocation id;

    @SubscribeEvent
    public static void init(LootTableLoadEvent event) {
        table = event.getTable();
        id = event.getName();

        modifyBlockLoot(ItemRegistry.DIAMOND_NUGGET.get(), 1f, 2f,"diamond_ore", "deepslate_diamond_ore");
        modifyBlockLoot(ItemRegistry.DIAMOND_NUGGET.get(), ConstantConfig.DIAMOND_NUGGET_FROM_COAL.get().floatValue(), 1f,
                "coal_ore", "deepslate_coal_ore");
    }

    private static void modifyEntityLoot(Item item, float chance, float maxAmount, String... entities) {
        for (String entity : entities) {
            if (id.equals(ConstantLootModifier.getEntityID(entity))) {
                table.addPool(ConstantLootModifier.createPool(item, chance, maxAmount).build());
            }
        }
    }

    private static void modifyBlockLoot(Item item, float chance, float maxAmount, String... blocks) {
        for (String block : blocks) {
            if (id.equals(ConstantLootModifier.getBlockID(block))) {
                table.addPool(ConstantLootModifier.createPool(item, chance, maxAmount).build());
            }
        }
    }

    private static void modifyStructureLoot(Item item, float chance, float maxAmount, String... structures) {
        for (String structure : structures) {
            if (id.equals(ConstantLootModifier.getChestID(structure))) {
                table.addPool(ConstantLootModifier.createPool(item, chance, maxAmount).build());
            }
        }
    }
}
````

## ConstantLootModifier Class
````java
public class ConstantLootModifier {

    public static LootPool.Builder createPool(Item item, float chance, float maxAmount) {
        return LootPool.lootPool()
                .setRolls(ConstantValue.exactly(1f))
                .when(LootItemRandomChanceCondition.randomChance(value(chance)))
                .add(LootItem.lootTableItem(item))
                .apply(SetItemCountFunction.setCount(UniformGenerator.between(1f, maxAmount)));
    }

    public static ResourceLocation getBlockID(String id) {
        return new ResourceLocation("blocks/" + id);
    }

    public static ResourceLocation getChestID(String id) {
        return new ResourceLocation("chests/" + id);
    }

    public static ResourceLocation getEntityID(String id) {
        return new ResourceLocation("entities/" + id);
    }

    private static float value(float chance) {
        return Math.min(1.0f, Math.max(0.0f, chance));
    }
}
````

**Description of issue:**
With Forge it worked properly, but on NeoForge nothing happens when using the LootTableEvent.
From what i can say the Event not really changed from Forge to NeoForge.

**Comment by Technici4n** [comment]
The event was fixed in 1.20.4, I recommend updating to that version.

**Comment by Stein-N** [comment]
Oh ok, tried to port my mod to 1.20.2 and was confused.
But thanks for the fast response :)
